### ISSUE TITLE
Dialyzer warnings in VS Code when creating a new project with --no-html

### ISSUE THREAD

**Issue by westmark** [phoenixframework/phoenix#5375]
### Environment

* Elixir version (elixir -v): 1.14.3
* Phoenix version (mix deps): 1.7.2
* Operating system: MacOS Ventura 13.2.1

### Expected behavior

```bash
> mix phx.new phx_test --no-html
> cd phx_test
> code .
```

After ElixirLS has done its thing, no warnings should be visible in the problems tab

### Actual behavior

Without editing anything, the Dialyzer reports two warnings in `lib/phx_test_web/router.ex`:

```elixir
use PhxTestWeb, :router      Function call/2 has no local return
```

At end of file
```elixir
end
    The pattern 
          {_@10, _@11, _@12, _@13} can never match the type 
          'error'
```

### Analysis

I have tried almost all permutations of `--no-xx` and only `--no-html` will generate these warnings whenever used. For the life of me I cannot find what differs in the generated code. I have even tried adding the only missing line in `def router` - `import Phoenix.LiveView.Router` (and required dependencies), but it does nothing. Removing everything but `def router` does not help either.

**Comment by chrismccord** [comment]
I don't have the bandwidth to look at this one but if someone wants to take a stab at making dialyzer happy with a PR, please let me know. Thanks!

**Comment by westmark** [comment]
@chrismccord I created a project without using `--no-html` to get going, but the warning popped up the second I removed the line `get "/", PageController, :home` from the `scope`. However, the warning will not appear If that line is kept in place and a new empty scope (with just a `pipe_through` ) is added. So whenever you only have empty scopes in the router, the warning will appear. I guess this will never happen in a real project(what would be the point?) so I'm totally fine with this being closed
